### PR TITLE
New readme file on how to build the documentation

### PR DESCRIPTION
**Pull request by mtmail**
It didn't fit into `CONTRIBUTING.md` and I wasn't sure if it should be a subpage on http://nominatim.org/release-docs/develop/ somewhere.

### CODE REVIEW

File: docs/How-to-build-and-preview-documentation.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+   ```
+
+   If you develop inside a Vagrant virtual machine:
+   * add port forwarding to your Vagrantfile, e.g. `config.vm.network "forwarded_port", guest: 8000, host: 8000`
+   * use `mkdocs serve --dev-addr 0.0.0.0:8000` because the default localhost
+      IP does not get forwarded.
+
+4. Push changes to the nominatim.org webserver
+
+   This step is manual and requires ssh access to the server.
```
Comment (lonvia) [inline docs/How-to-build-and-preview-documentation.md]: No it's not. The master branch is automatically deployed every night on under https://nominatim.org/release-docs/develop/ I suggest to remove point 4 completely and just mention the fact.

Comment (mtmail) [reply docs/How-to-build-and-preview-documentation.md]: removed

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ theme: readthedocs
 docs_dir: ${CMAKE_CURRENT_BINARY_DIR}
 site_url: http://nominatim.org
 repo_url: https://github.com/openstreetmap/Nominatim
-pages:
+nav:
```
Comment (lonvia) [inline docs/mkdocs.yml]: I know that newer versions of mkdocs are complaining but I'd like to leave that for a while to remain backwards compatible.

Comment (mtmail) [reply docs/mkdocs.yml]: sure, removed now/again

**Pull request merged**